### PR TITLE
Add Remaining Forks Needed By Temporalx (WIP)

### PR DESCRIPTION
**Pull request by bonedaddy**
Adds the remaining forks needed by TemporalX. Some new tests have been disabled which is fine, the goal of this is to get the remaining repos forked that we need so we can trial using this library in dev.

### CODE REVIEW

**Comment by bonedaddy** [comment]
Interesting errors that have shown up:
```
?   	github.com/RTradeLtd/libp2px/pkg/pnet	[no test files]
==6311==ERROR: ThreadSanitizer failed to allocate 0x80000 (524288) bytes of clock allocator (error code: 12)
FATAL: ThreadSanitizer CHECK failed: ./gotsan.cc:6976 "((0 && "unable to mmap")) != (0)" (0x0, 0x0)
FAIL	github.com/RTradeLtd/libp2px/pkg/pubsub	372.222s
```

**Comment by bonedaddy** [comment]
I  think some of the timeouts in the pubsub tests are due to the lack of a cancellable context in the peerstore. See this issue for more information https://github.com/RTradeLtd/libp2px/issues/10

File: pkg/http/p2phttp.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+//	res, err := client.Get("libp2p://Qmaoi4isbcTbFfohQyn28EiYM5CDWQx9QRCjDh3CTeiY7P/hello")
+//	...
+//
+// In the example above, the client registers a "libp2p" protocol for which the
+// custom transport is used. It can still perform regular "http" requests. The
+// protocol name used is arbitraty and non standard.
+//
+// Note that LibP2P hosts cannot dial to themselves, so there is no possibility
+// of using the same host as server and as client.
+package p2phttp
```
Comment (xiegeo) [inline pkg/http/p2phttp.go]: I suggest renaming the folder to p2phttp too.

**Comment by bonedaddy** [comment]
gossipsub race detected:

```
==================
WARNING: DATA RACE
Read at 0x00000159b9a8 by goroutine 2622:
  github.com/RTradeLtd/libp2px/pkg/pubsub.(*GossipSubRouter).heartbeat()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/gossipsub.go:476 +0xd8b
  github.com/RTradeLtd/libp2px/pkg/pubsub.(*GossipSubRouter).heartbeat-fm()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/gossipsub.go:423 +0x41
  github.com/RTradeLtd/libp2px/pkg/pubsub.(*PubSub).processLoop()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/pubsub.go:476 +0xdc8
Previous write at 0x00000159b9a8 by goroutine 1851:
  github.com/RTradeLtd/libp2px/pkg/pubsub.TestGossipsubFanoutExpiry()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/gossipsub_test.go:262 +0x52
  testing.tRunner()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:909 +0x199
Goroutine 2622 (running) created at:
  github.com/RTradeLtd/libp2px/pkg/pubsub.NewPubSub()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/pubsub.go:239 +0x1270
  github.com/RTradeLtd/libp2px/pkg/pubsub.NewGossipSub()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/gossipsub.go:49 +0x33d
  github.com/RTradeLtd/libp2px/pkg/pubsub.getGossipsubs()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/gossipsub_test.go:15 +0x15d
  github.com/RTradeLtd/libp2px/pkg/pubsub.TestGossipsubFanoutMaintenance()
      /home/travis/gopath/src/github.com/RTradeLtd/libp2px/pkg/pubsub/gossipsub_test.go:186 +0x14f
  testing.tRunner()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:909 +0x199
Goroutine 1851 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:909 +0x199
  testing.runTests()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.13.6.linux.amd64/src/testing/testing.go:1117 +0x2ff
  main.main()
      _testmain.go:224 +0x337
==================
--- FAIL: TestGossipsubFanoutExpiry (5.80s)
    testing.go:853: race detected during execution of test
```

**Pull request merged**